node/src/main/scala/coop/rchain/node/configuration/Configuration.scala

```diff
@@ -49,24 +49,74 @@ object Configuration {
 
   def apply(arguments: Seq[String])(implicit log: Log[Task]): Task[Configuration] =
     for {
-      options    <- Task.delay(commandline.Options(arguments))
-      profile    <- Task.pure(options.profile.toOption.flatMap(profiles.get).getOrElse(defaultProfile))
-      _          <- log.info(s"Starting with profile ${profile.name}")
-      dataDir    <- Task.pure(options.run.data_dir.getOrElse(profile.dataDir._1()))
-      configFile <- Task.delay(options.configFile.getOrElse(dataDir.resolve("rnode.toml")).toFile)
-      _          <- log.info(s"Using configuration file: $configFile")
-      configE    <- Task.delay(toml.TomlConfiguration.from(configFile))
-      _ <- configE match {
-            case Right(_) => Task.unit
-            case Left(e)  => log.warn(s"Can't load the configuration file: $e")
-          }
-      config <- Task.pure(configE.toOption)
-      effectiveDataDir <- Task.pure(
-                           if (options.run.data_dir.isDefined) dataDir
-                           else config.flatMap(_.server.flatMap(_.dataDir)).getOrElse(dataDir))
-      result <- Task.pure(apply(effectiveDataDir, options, config))
+      options <- Task.delay(commandline.Options(arguments))
+      profile <- Task.pure(options.profile.toOption.flatMap(profiles.get).getOrElse(defaultProfile))
+      _       <- log.info(s"Starting with profile ${profile.name}")
+      result  <- apply(options, subcommand(options), profile)
     } yield result
 
+  private def apply(options: commandline.Options, command: Command, profile: Profile)(
+      implicit log: Log[Task]): Task[Configuration] =
+    if (command == Run) {
```

Comment:
I understand the context. This is one of those (rare) places, were maybe a comment is neded?